algebird-core/src/main/scala/com/twitter/algebird/SetDiff.scala

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2016 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.twitter.algebird
+
+/**
+ * `SetDiff` is a class that represents changes applied to a set. It
+ * is in fact a Set[T] => Set[T], but doesn't extend Function1 since
+ * that brings in a pack of methods that we don't necessarily want.
+ */
+sealed abstract case class SetDiff[T] private (add: Set[T], remove: Set[T]) { self =>
+  def +(t: T): SetDiff[T] = SetDiff(add + t, remove - t)
+  def -(t: T): SetDiff[T] = SetDiff(add - t, remove + t)
+  def ++(ts: Iterable[T]): SetDiff[T] = SetDiff(add ++ ts, remove -- ts)
+  def --(ts: Iterable[T]): SetDiff[T] = SetDiff(add -- ts, remove ++ ts)
+  def merge(other: SetDiff[T]): SetDiff[T] = {
+    val newAdd = (add ++ other.add) -- other.remove
+    val newRem = (remove -- other.add) ++ other.remove
+    SetDiff(newAdd, newRem)
+  }
+
+  /**
+   * Applies the contained diff to the supplied set.
+   */
+  def apply(previous: Set[T]): Set[T] = previous ++ add -- remove
+
+  /**
+   * Returns a diff that, if applied to a set, undoes the effects of
```

Comment:
May want to be extra careful here, since this is looking very much like a `Bijection[Set[T], Set[T]]`, though of course it would not obey the `Bijection` laws.